FifthLesson/VectorProgram/VectorProgram/Model/Vector.cs

```diff
@@ -1,115 +1,117 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using VectorProgram.Controller;
 
 namespace VectorProgram.Model
 {
-    public sealed class Vector
+    public class Vector : IEquatable<Vector>
     {
-        public double CoordinateFirst { get; private set; }
-        public double CoordinateSecond { get; private set; }
-        public double CoordinateThird { get; private set; }
-
+        public double FirstCoordinate { get; private set; }
+        public double SecondCoordinate { get; private set; }
+        public double ThirdCoordinate { get; private set; }
         public double Length
         {
-            get
-            {
-                return Math.Round(Math.Sqrt(CoordinateFirst * CoordinateFirst + CoordinateSecond * CoordinateSecond + CoordinateThird * CoordinateThird), 3);
-            }
+            get => Math.Round(Math.Sqrt(Math.Pow(FirstCoordinate, 2) + Math.Pow(SecondCoordinate, 2) + Math.Pow(ThirdCoordinate, 2)), 3);
         }
 
-        public Vector()
-        {
-            CoordinateFirst = 0;
-            CoordinateSecond = 0;
-            CoordinateThird = 0;
-        }
+        public Vector() { }
 
-        public Vector(double coordinateFirst, double coordinateSecond, double coordinateThird) : this()
+        public Vector(double coordinateFirst, double coordinateSecond, double coordinateThird)
         {
-            CoordinateFirst = coordinateFirst;
-            CoordinateSecond = coordinateSecond;
-            CoordinateThird = coordinateThird;
+            FirstCoordinate = coordinateFirst;
+            SecondCoordinate = coordinateSecond;
+            ThirdCoordinate = coordinateThird;
         }
+        
+        public Vector(Vector vector) : this(vector.FirstCoordinate, vector.SecondCoordinate, vector.ThirdCoordinate) { }
 
-        public Vector(Vector vector) : this()
+        public double CalculateAngleBetweenVectors(Vector vector)
         {
-            CoordinateFirst = vector.CoordinateFirst;
-            CoordinateSecond = vector.CoordinateSecond;
-            CoordinateThird = vector.CoordinateThird;
+            var angle = VectorHelper.CalculateAngleBetweenVectors(this, vector);
+            return Math.Round(angle, 3);
         }
 
-        public override string ToString()
-        {
-            return $"({CoordinateFirst}, {CoordinateSecond}, {CoordinateThird})";
-        }
+        public double CalculateModuleVector(Vector vector) => VectorHelper.CalculateModuleVector(vector);
+        
+        public override string ToString() => $"({FirstCoordinate}, {SecondCoordinate}, {ThirdCoordinate})";
 
-        public static Vector operator +(Vector vector1, Vector vector2)
+        public override int GetHashCode()
```

Comment:
It's really good idea how to calculate hash code, but, please, simplify it. Try to avoid using of any cycle

Reply:
Removed cycle.